File: 백준/Silver/3230. 금메달， 은메달， 동메달은 누가？/금메달， 은메달， 동메달은 누가？.cc
#include <iostream>
#include <vector>

using namespace std;

int main(){
    int N,M;
    cin>>N>>M;
    
    vector <int> firstTable;
    for (int i=0;i<N;i++){
        int rank;
        cin>>rank;
        firstTable.insert(firstTable.begin()+rank-1, i);

    }
    vector <int> secondTable;
    for (int i=0;i<M;i++){
        int rank;
        cin>>rank;
        secondTable.insert(secondTable.begin()+rank-1,firstTable[M-1-i]);
    }
    for (int i=0;i<3;i++){
        cout<<secondTable[i]+1<<'\n';
    }
    return 0;
}

File: 백준/Silver/3230. 금메달， 은메달， 동메달은 누가？/금메달， 은메달， 동메달은 누가？.py
N,M = map(int,input().split())

first = [int(input()) for _ in range(N)]

second = [int(input()) for _ in range(M)]

table = []

for i in range(N):
    table.insert(first[i]-1,i)
secondTable=[]
for i in range(M-1,-1,-1):
    secondTable.insert(second[M-1-i]-1,table[i])
for a in secondTable[:3]:
    print(a+1)